src/map/skill.cpp

```diff
@@ -16200,6 +16298,31 @@ int skill_unit_onplace_timer(struct skill_unit *unit, struct block_list *bl, t_t
 				skill_attack(skill_get_type(sg->skill_id), ss, &unit->bl, bl, sg->skill_id, sg->skill_lv, tick, flag);
 			}
 			break;
+			
+		case UNT_TOTEM_OF_TUTELARY:
+			++sg->val1;
+			if( bl->type == BL_PC ) {
+				if (sg->val1 % 3 == 0) {
+					int hp, sp, ap;
+
+					hp = 1000 + 500 * sg->skill_lv + 5 * tstatus->crt + 7 * status_get_lv(bl) + 50 * pc_checkskill(tsd, SOA_TALISMAN_MASTERY);
+					sp = 50 + 50 * sg->skill_lv + 5 * tstatus->crt + 7 * status_get_lv(bl) + 5 * pc_checkskill(tsd, SOA_TALISMAN_MASTERY);
+					ap = 0;
+					
+					if (tstatus->hp < tstatus->max_hp)
```

Comment:
Would it be better to split these into 3x status_heal calls? And use the return value for clif_skill_damage?
Because like this it will not check if the target has some status changes that prevent healing etc.